WcaOnRails/app/models/competition.rb

```diff
@@ -266,11 +294,37 @@ def registration_must_close_after_it_opens
     end
   end
 
+  def use_wca_as_website=(value)
+    @use_wca_as_website = ActiveRecord::Type::Boolean.new.type_cast_from_database(value)
+  end
+
+  def use_wca_as_website
+    @use_wca_as_website || website == url
+  end
+
+  before_validation :assign_wca_website_url
+  private def assign_wca_website_url
+    if use_wca_as_website && website != url
+      self.website = self.url
+    end
+  end
+
+  after_save :update_website
+  private def update_website
+    if use_wca_as_website && website != url
+      update_column :website, url
```

Comment:
Basically there were two points I wanted to achieve:
- always store the website in the db
- don't use another column that isn't necessary to make things working
But I think you're right and it's not worth its trickiness.
I'll update the code when I'm back =)